pkg/github/teamreadwriter.go

```diff
@@ -277,6 +280,28 @@ func (g *TeamReadWriter) getGitHubUser(ctx context.Context, client *github.Clien
 	return user, nil
 }
 
+// GetGitHubOrgSaml gets the saml identities for the github org.
+// If the saml for the given orgID is expired in cache or does not exisit,
+// it will retrieve the newest saml identitiy information.
+func (g *TeamReadWriter) GetGitHubOrgSaml(ctx context.Context, orgID int64) (map[string]struct{}, error) {
+	orgIDStr := strconv.FormatInt(orgID, 10)
+	if orgSaml, ok := g.orgSamlIdentitiesCache.Lookup(orgIDStr); ok {
+		return orgSaml, nil
+	}
+	logger := logging.FromContext(ctx)
+	logger.InfoContext(ctx, "fetching org saml identities", "org_id", orgID)
+	gqlClient, err := g.graphqlClientForOrg(ctx, orgID, g.endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create graphQL client: %w", err)
+	}
+	saml, err := GetOrgSamlIdentitiesByOrgID(ctx, g.client, gqlClient, orgID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch org saml identities: %w", err)
+	}
+	g.orgSamlIdentitiesCache.Set(orgIDStr, saml)
```

Comment:
We might consider using https://pkg.go.dev/github.com/abcxyz/pkg/cache#Cache.WriteThruLookup rather than Lookup() and Set()